### PR TITLE
Added support for scp

### PR DESCRIPTION
**Pull request by vidarl**
So, you currently have support for `ssh` and `mount:upload` and `mount:download`, but I find it more convenient if we could use more or less the scp syntax. Besides, the `mount` commands works on directories, making it impossible to upload/download single files.

The whole point with the `scp` command is to avoid  having to fetch the ssh url all the time, ie:

```
scp a.txt foobar2123123-master-7rqtwti--app@ssh.eu-2.platform.sh:var/upload
```

Instead, it is a simple as:
```
#upload
platform scp a.txt remote:var/upload

#upload multiple files
platform scp a.txt b.txt c.txt remote:var/upload


#download
platform scp remote:var/upload/a.txt .

```

I am aware of the  workaround given below, but I think supporting scp directly is way better
```
scp "$(platform ssh --pipe)":var/upload/a.txt .
```

### CODE REVIEW

File: src/Command/Environment/EnvironmentScpCommand.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        /** @var \Platformsh\Cli\Service\Scp $scp */
+        $scp = $this->getService('scp');
+        $scpOptions = [];
+        $command = $scp->getScpCommand($scpOptions);
+
+        if ($input->getOption('recursive')) {
+            $command .= ' -r';
+        }
+
+        if ($input->getOption('quiet')) {
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentScpCommand.php]:
```suggestion
        if ($output->isVerbose()) {
            $command .= ' -v';
        } elseif ($output->isQuiet()) {
```

File: src/Command/Environment/EnvironmentScpCommand.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            $command .= ' -q';
+        }
+
+        $remoteUsed = false;
+        foreach ($files as $key => $file) {
+            if (strncasecmp($file, 'remote:', '7') === 0) {
```
[the remark this answers is not in the document]
Comment (vidarl) [reply src/Command/Environment/EnvironmentScpCommand.php]: Sorry for the delay, for whatever reason, I have missed the notification emails for your suggestions...

Anyway, reason for my use of `strncasecmp` was to some people fancy uppercacse (`REMOTE:.....`). Only supporting lower case is fine by me

**Comment by vidarl** [comment]
Thank you for your feedback. You want me to squash the commits?

**Comment by vidarl** [comment]
@pjcdawkins: Any chance this PR could get some love now and be merged?

**Comment by pjcdawkins** [comment]
Oops sorry for the neglect.

I realise I was uneasy about the service structure. Instead of the `scp` service, could getScpCommand() be a method on the `ssh` service? Everything the scp service is currently doing relates to ssh, it's not doing anything scp-specific.

Alternatively, move some of the scp-specific logic (currently in the command's `execute()`) into the `scp` service. Maybe it could depend on `ssh`.

**Comment by pjcdawkins** [comment]
I don't mind doing that refactoring myself if you'd rather focus time elsewhere. Functionally this is great.

**Comment by vidarl** [comment]
I just want this in as I miss it every day. I won't have time for this until later this week or next. But feel free to refactor if you are available..

**Comment by vidarl** [comment]
Thank you. I assume refactoring regarding dedicated `scp` service is not a topic after all then.

**Pull request merged**